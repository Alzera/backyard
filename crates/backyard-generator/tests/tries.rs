use backyard_generator::generate;
use backyard_parser::parse_eval;

#[test]
fn basic() {
  let asts = parse_eval("try {
} catch (Exception $e) {
}").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn multiple_types() {
  let asts = parse_eval("try {
} catch (UnknownGetterException | ReflectionException) {
}").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn finally() {
  let asts = parse_eval(
    "try {
  throw new Error(\"Custom error occurred\");
} catch (FooError $err) {
} catch (Foo2Error | BarError $err) {
} finally {
}"
  ).unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}
