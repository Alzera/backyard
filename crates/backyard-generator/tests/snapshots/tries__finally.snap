---
source: crates/backyard-generator/tests/tries.rs
expression: generate(asts).unwrap()
snapshot_kind: text
---
"try {\n  throw new Error(\"Custom error occurred\");\n} catch (FooError $err) {\n} catch (Foo2Error | BarError $err) {\n} finally {\n}"
